Automattic/woocommerce-payments
# Analytics: Ensure the store default currency always displays in list

Fixes #4479 

#### Changes proposed in this Pull Request

* Add the default currency to the currency list that is used for the Customer Currency filter in Analytics as it's cached for 5 minutes. 

**Before**
<img width="1596" alt="image" src="https://user-images.githubusercontent.com/79862886/183877288-282ee137-f285-4f90-9ed3-d2d9947cc35d.png">
**After**
<img width="1596" alt="image" src="https://user-images.githubusercontent.com/79862886/183877344-7b9e3e41-335d-4a89-8eb2-cbe27620d024.png">

#### Testing instructions

1. Launch a new JN site. You can use specialops page for that, just include WooCommerce, WCPay Dev Tools and WooCommerce Payments and tick Launch single site.
2. Open the newly created site and complete WooCommerce onboarding. Choose Storefront theme so it will be convenient to test the fix.
3. Visit the `Analytics -> Orders` page, and try to Add the Customer Currency filter. You shouldn’t be able to see select input with the currencies.
<img width="1596" alt="image" src="https://user-images.githubusercontent.com/79862886/183869267-bcb9bb87-87e0-4f1d-9569-b49660aa8701.png">

4. Visit the `/wp-admin/admin.php?page=wcpaydev` page to see the server credentials.
5. SSH to the server using the SSH command from WCPay Dev
6. Navigate to the plugin folder `cd apps/userXXX/public/wp-content/plugins/woocommerce-payments`. 
7. Modify `register_customer_currencies()` in `includes/multi-currency/Analytics.php` by adding the code below after `$default_currency = $this->multi_currency->get_default_currency();`
`// Add default currency to the list if it does not exist.`
`if ( ! in_array( $default_currency->get_code(), $currencies ) ) {`
    `$currencies[] = $default_currency->get_code();`
`}`

8. Visit the `Analytics -> Orders` page, and try to Add the Customer Currency filter. Now you should be able to see select input with USD currency.
9. Setup multi-currency by adding GBP. You’ll need to complete Stripe onboarding prior to that. Tick `Add a currency switcher to the Storefront theme on the breadcrumb section.` so it'll be easy to switch between currencies.
10. Comment out the code you added in step 7. 
11. Visit the Store and create orders in USD and GBP.
12. Visit the `Analytics -> Orders` page, you should see the orders but the Customer Currency list is empty.
<img width="1593" alt="image" src="https://user-images.githubusercontent.com/79862886/183874815-0c0762ac-3846-4729-b1cb-6fab33c7c5d1.png">
13. Uncomment the code and see that now you can see only USD in the currency list.
<img width="1579" alt="image" src="https://user-images.githubusercontent.com/79862886/183413007-0b8410ed-06ae-48bc-b3e7-6ebd9aa83a41.png">
14. Wait 5 minutes and see that now you are able to see Pound Sterling and USD in the Customer Currency list.
<img width="1579" alt="image" src="https://user-images.githubusercontent.com/79862886/183875089-722048c0-d999-48be-b9f2-70f4edfab62f.png">


-------------------


**Post merge**
